docs/DeveloperGuide.md

```diff
@@ -1,34 +1,503 @@
+|v1.0|Student who likes to keep notes on places I travel before|To note down my fresh experiences while on travel trips|Open and look back at it anytime|
+|v1.0|Student who is always on-the-go|To input my spendings in a very quick manner|Input my spendings without a hassle|
+|v2.0|Student traveling to a foreign country for the first time|To record cultural differences and some other travel notes before traveling there|Access them quickly in that country to avoid any misunderstandings|
+|v2.0|Student unsure on which country to visit for exchange | To plan various trips before looking at them individually | Make an informed decision which country I'm most interested in for exchange|
+|v2.0|Student who is not very good at managing expenses|To be reminded when my spendings are going to exceed my initial budget|Reduce my spendings if required|
+|v2.0|Student who plans trip ahead of time|To be able to edit whatever has been planned easily|Improve my itinerary easily|
+|v2.0|Student who is interested in many overseas opportunities|To create ane edit multiple trips simultaneously|Plan multiple trips ahead at a time|
+
+### 5.3 Use Cases
+
+Here are the list of use cases for Trippie.
+
+1. Don't have a tool to plan trips.
+2. Use Trippie to plan trips.
+3. Able to create trips easily.
+4. Share the trip files to others.
+5. Be a happy student.
+
+### 5.4 Non-Functional Requirements
+
+| Requirement Type | Description |
+|------------------|-------------|
+|Constraint|Single user product|
+|Performance|Software should not be dependent on a remote server|
+|Performance|Software should not exceed 100Mb for JAR file and 15MB per PDF file|
+|Quality|Users should prefer CLI/Typing|
+|Technical|Must have Java 11 installed|
+|Technical|No DBMS, all data to be stored locally|
+|Technical|Data stored must be in human-editable files|
+|Technical|Programme should be platform independent|
+|Technical|Programme should work without an installer|
+
+### 5.5 Glossary
+
+* *CLI* - Command Line Interface
+* *IntelliJ* - A Java IDE developed by Jetbrains
+* *JUnit* - Java Unit Testing Framework
+* *Java* - A programming language
+
+## 6.0 Appendix: Instructions for Manual Testing
+
+<kbd>
+
+![UML Diagram Main](https://github.com/AY2021S1-CS2113T-W11-2/tp/blob/master/docs/Trippie%20UML-Main.jpg?raw=true)
+<center><i>Figure 9: Overall Architecture</i></center>
+</kbd>
+
+![UML Diagram](https://github.com/AY2021S1-CS2113T-W11-2/tp/blob/master/docs/Trippie%20UML-Class%20Diagram.jpg?raw=true)
+<center><i>Figure 10: Class Types and Parameters</i></center>
+
+![UML Diagram Command](https://github.com/AY2021S1-CS2113T-W11-2/tp/blob/master/docs/Trippie%20UML-Command.jpg?raw=true)
+<center><i>Figure 11: Commands Classes</i></center>
+
+![UML Diagram Exception](https://github.com/AY2021S1-CS2113T-W11-2/tp/blob/master/docs/Trippie%20UML-Exception.jpg?raw=true)
+<center><i>Figure 12: Exception Classes</i></center>
```

Comment:
Should this UML diagrams be under instructions for manual testing? Probably would be better as an overview rather than instructions?